pkgs/applications/editors/imhex/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib
+, gcc12Stdenv
+, cmake
+, llvm
+, fetchFromGitHub
+, mbedtls
+, gtk3
+, pkg-config
+, capstone
+, dbus
+, libGLU
+, glfw3
+, file
+, perl
+, python3
+, jansson
+, curl
+, fmt_8
+, nlohmann_json
+, xlibsWrapper
+, yara
+}:
+
+let
+  # when bumping the version, check if imhex has gotten support for the capstone version in nixpkgs
+  version = "1.19.3";
+
+  patterns_src = fetchFromGitHub {
+    owner = "WerWolv";
+    repo = "ImHex-Patterns";
+    rev = "ImHex-v${version}";
+    hash = "sha256-mukGPN2TugJZLLuZ5FTvZ4DxUsMGfVNhBFAPnBRC0qs=";
+  };
+
+in
+gcc12Stdenv.mkDerivation rec {
+  pname = "imhex";
+  inherit version;
+
+  src = fetchFromGitHub {
+    fetchSubmodules = true;
+    owner = "WerWolv";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-SFv5ulyjm5Yf+3Gpx+A74so2YClCJx1sx0LE5fh5eG4=";
+  };
+
+  nativeBuildInputs = [ cmake llvm python3 perl pkg-config xlibsWrapper ];
+
+  buildInputs = [
+    capstone
+    curl
+    dbus
+    file
+    fmt_8
+    glfw3
+    gtk3
+    jansson
+    libGLU
+    mbedtls
+    nlohmann_json
+    yara
+  ];
```

Comment:
Some of them are not found by `rg -i ...` in sources such as jansson. Are they necessary? OpenGL and freetype included by cmake `find_package` are not listed here.  They may be input from other dependencies, but it is better to add it here explicitly.